ccm_web/server/src/canvas/canvas.interfaces.ts

```diff
@@ -66,14 +66,38 @@ export interface CanvasCourseSection extends CanvasCourseSectionBase {
   total_students: number
 }
 
-export enum UserEnrollmentType {
+export enum CanvasRoles {
+  AccountAdmin = 'Account Admin',
+  SubAccountAdmin = 'Sub-Account Admin',
+  SupportConsultant = 'Support Consultant',
+  Assistant = 'Assistant',
   DesignerEnrollment = 'DesignerEnrollment',
   ObserverEnrollment = 'ObserverEnrollment',
   StudentEnrollment = 'StudentEnrollment',
   TaEnrollment = 'TaEnrollment',
   TeacherEnrollment = 'TeacherEnrollment'
 }
 
+export enum UserEnrollmentType {
+  DesignerEnrollment = CanvasRoles.DesignerEnrollment,
+  ObserverEnrollment = CanvasRoles.ObserverEnrollment,
+  StudentEnrollment = CanvasRoles.StudentEnrollment,
+  TaEnrollment = CanvasRoles.TaEnrollment,
+  TeacherEnrollment = CanvasRoles.TeacherEnrollment
+}
+
+// valid role types for LTI launch
+// as defined in https://docs.google.com/spreadsheets/d/1pm5y9FX0zrDX7Zy3mOyDLxmA-iKoWfmlxvbtNkWg6Zw/edit#gid=1360549473
+export enum LTIEnrollmentTypes {
+  AccountAdmin = CanvasRoles.AccountAdmin,
+  SubAccountAdmin = CanvasRoles.SubAccountAdmin,
+  SupportConsultant = CanvasRoles.SupportConsultant,
+  TeacherEnrollment = CanvasRoles.TeacherEnrollment,
+  DesignerEnrollment = CanvasRoles.DesignerEnrollment,
+  TaEnrollment = CanvasRoles.TaEnrollment,
+  Assistant = CanvasRoles.Assistant
+}
```

Comment:
This seems like it would work but also feels kinda repetitive. Did you consider using arrays of enum member values instead? For example:
```
AllowedEnrollmentTypes = [CanvasRoles.DesignerEnrollment, CanvasRoles.ObserverEnrollment, CanvasRoles.StudentEnrollment, CanvasRoles.TaEnrollment, CanvasRoles.TeacherEnrollment]
```
You'd have to tweak the enrollment API endpoint validation, since I think it uses an enum right now, but that's not an insurmountable task. Just a thought.

Reply:
I guess we can postpone this for now. I started looking at it, not impossible, but it may result in the typing on `CanvasEnrollment` becoming a bit relaxed (just a string, instead of an enum member). We can go with this for now and maybe refactor later when an opportunity arises.